Boilerplate windows support

denis256 opened the pull request:
<!-- Prepend '[WIP]' to the title if this PR is still a work-in-progress. Remove it when it is ready for review! -->

## Description
Included changes:
* Fixed handling of paths in Windows
* Added support for Windows CICD
* Fixed base tests for Windows environment
* Changed "kebab test" to generate files with placeholder instead of saving files in git since it breaks repo clone in Windows
* Added skipping for some tests that relay on OS native commands
* Updated mime type detection to use only read data

Before this change:

<img width="1592" alt="Screenshot 2024-03-04 at 20 48 49" src="https://github.com/gruntwork-io/boilerplate/assets/10694338/f7ac136e-5616-45c7-a4be-afe8def92212">


With the change from this PR:

<img width="1353" alt="Screenshot 2024-03-04 at 20 46 25" src="https://github.com/gruntwork-io/boilerplate/assets/10694338/0a4df8de-eddd-4c4f-97e7-bd3a0d4b71f1">



<!-- Description of the changes introduced by this PR. -->

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

<!-- One-line description of the PR that can be included in the final release notes. -->
Added / Removed / Updated [X].

Updated boilerplate template path identification in Windows.

### Migration Guide

<!-- Important: If you made any backward incompatible changes, then you must write a migration guide! -->

brikis98 commented:
This is the announcement blog post I was talking about: https://blog.gruntwork.io/introducing-boilerplate-6d796444ecf6. It contains links to the boilerplate repo, but I made the links go to a specific commit, so if you make changes to the example code, it shouldn't break anything. We really _do_ need the dynamic name in the test file (hence the `{{ xxx }}` in the path), but fixing all the issues on Windows is pretty important, so perhaps for now, you can update that example to dynamically generate the file, so we can get this merged soon, and we can come back to a more long-term fix in the future?

denis256 commented:
Fixed tests for Windows and updated approach with variables in file names - in git are stored "generic" names and in tests for "unix" platform files are renamed with variables in the name

https://github.com/gruntwork-io/boilerplate/pull/164/files#diff-029441898e3ff4247efb42bd1056bce6b2b51b6a8950731f9b94c71ff93ae364R78

brikis98 reviewed (approved):
LGTM! Please merge and release.
